test/abstract_syntax_tree_location_data.coffee

```diff
@@ -0,0 +1,392 @@
+# Astract Syntax Tree location data
+# ---------------------------------
+
+testAstLocationData = (code, expected) ->
```

Comment:
Refactored these to support nested location data testing. Preserved the existing custom error messaging for location data fields (with additional "path" info to help identify which node was failing)